examples/c-guest/guest/riscv32im-risc0-zkvm-elf.ld

```diff
@@ -0,0 +1,77 @@
+/*
+  Copyright 2023 Risc0, Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+OUTPUT_FORMAT("elf32-littleriscv", "elf32-littleriscv", "elf32-littleriscv")
+OUTPUT_ARCH(riscv)
+ENTRY(_start)
+
+/* Must match risc0/zkvm/platform/src/memory.rs */
+/* TODO double check these memory sections, probably not ideal */
+MEMORY {
+  stack        : ORIGIN = 0x00000400, LENGTH =  8M - 1K
+  data    (RW) : ORIGIN = 0x00080000, LENGTH =  24M
+  heap         : ORIGIN = 0x02000000, LENGTH =  80M
+  prog    (X)  : ORIGIN = 0x07000000, LENGTH =  80M
+}
```

Comment:
I'd love to have this be resolved before this comes in. Seems like the Rust memory is:
```
stack (0x400 - 0x0020_0400) | (start 0x0020_0800) prog/.text | .data and .bss shared (top 0x0C00_0000)
```

Would it be reasonble to match this instead, and have these sections be more dynamic rather than fixed as they are here? Ordering is also diff with prog above the heap, which may not have the same checks (seems like the allocator just checks if memory is above the system range, but this can allocate over the program?)

Reply:
I'm not sure how to match the rust model with a linker script, because the heap starts right at the end of text. At least in C, it should be possible to rely on the `_end` symbol to exist (I think most linkers define it).